bin/install-pr

```diff
@@ -9,63 +9,138 @@
 then
   # shellcheck disable=SC2016
-  echo 'Generate a personal access token at https://github.com/settings/tokens and set it in the $GITHUB_TOKEN env var'
+  echo 'Error: Generate a personal access token at https://github.com/settings/tokens and set it in the $GITHUB_TOKEN env var'
   exit 1
 fi
 
-auth="Authorization: token $GITHUB_TOKEN"
-
-branch=$(curl -s -H "$auth" "https://api.github.com/repos/linkerd/linkerd2/pulls/$pr" | jq -r '.head.ref')
-
-artifacts=$(curl -s -H "$auth" "https://api.github.com/repos/linkerd/linkerd2/actions/workflows/kind_integration.yml/runs?branch=$branch" | jq -r '.workflow_runs[0].artifacts_url')
+linkerd2_pulls_url="https://api.github.com/repos/linkerd/linkerd2/pulls"
+linkerd2_kind_integration_url="https://api.github.com/repos/linkerd/linkerd2/actions/workflows/kind_integration.yml"
 
-archive_url=$(curl -s -H "$auth" "$artifacts" | jq -r '.artifacts[0].archive_download_url')
+# Get the URL for downloading the artifacts archive
+auth="Authorization: token $GITHUB_TOKEN"
+branch=$(curl -sL -H "$auth" "$linkerd2_pulls_url/$pr" | jq -r '.head.ref')
+artifacts=$(curl -sL -H "$auth" "$linkerd2_kind_integration_url/runs?branch=$branch" | jq -r '.workflow_runs[0].artifacts_url')
+archive=$(curl -sL -H "$auth" "$artifacts" | jq -r '.artifacts[0].archive_download_url')
 
+bindir=$( cd "${BASH_SOURCE[0]%/*}" && pwd )
 dir=$(mktemp -d -t "linkerd-pr-$pr.XXXXXXXXXX")
 
+# Change directory temporarily and download artifact
 cd "$dir" || exit
 
-echo "### Downloading images ###"
-
-curl -L -o archive.zip -H "$auth" "$archive_url"
-
+curl -L -o archive.zip -H "$auth" "$archive"
 unzip -o archive.zip
 
-echo "### Pushing images ###"
+# Push the Linkerd CLI
+image=$(docker load -i cli-bin.tar | sed 's/Loaded image: //')
+docker push "$image"
+tag=$(echo "$image" | cut -f 2 -d ":")
 
-for archive in cli-bin cni-plugin controller debug grafana proxy web
-do 
-  image=$(docker load -i "$archive.tar" | sed "s/Loaded image: //")
-  docker push "$image"
-  tag=$(echo "$image" | cut -f 2 -d ":")
-done
+# Load images into local docker images
+if [ $is_kind ]
+then
+  # Create new KinD cluster
+  cluster_name="pr-$pr"
+  context_flag=(--context kind-"$cluster_name")
+
+  if "$bindir"/kind get clusters | grep "$cluster_name"
+  then
+    echo "Error: KinD cluster '$cluster_name' exists"
+    exit 1
+  fi
+  
+  "$bindir"/kind create cluster --name "$cluster_name"
+
+  # Load images into cluster
+  for image in cni-plugin controller debug grafana proxy web
+  do
+    echo "Loading $image image into cluster.."
+    "$bindir"/kind load image-archive --name "$cluster_name" $image.tar || tee load_fail &
+  done
+
+  # Wait for `kind load` background processes to copmlete; exit early if any
+  # job failed
+  wait < <(jobs -p)
+  if [ -f load_fail ]
+  then
+    echo "Loading docker images into KinD cluster failed."
+    exit 1
+  fi
+else
+  for image in cni-plugin controller debug grafana proxy web
+  do
+    docker load -i "$image.tar"
```

Comment:
don't we need to push these docker images?

Reply:
Because the images are loaded into the local docker images, the install pulls them from there instead of needing to look in the registry.

Reply:
if installing into a non-local (cloud) cluster, the images need to be pushed to a repository that the cluster can access

Reply:
Ah that's right yea I was thinking about local clusters.

We should only need to `docker push` for the non-KinD case so I'll add it back for that.